charts/generic-service/templates/alerts-logs.yaml

```diff
@@ -1,4 +1,29 @@
 {{- if and .Values.alerting.enabled .Values.alerting.logs.levels }}
+{{- if .Values.alerting.logs.lokiSideCar }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "generic-service.fullname" . }}
+  labels: {{- include "generic-service.top-level-labels" . | nindent 4 }}
+    {{ .Values.alerting.logs.lokiSideCar.configMapLabel }}: {{ .Values.alerting.logs.lokiSideCar.configMapLabelValue | quote }}
+
+data:
+  {{ include "generic-service.fullname" . }}.yaml: |
+    groups:
+      - name: {{ include "generic-service.fullname" . }}.logs
+        rules:
+          {{- range $level, $options := .Values.alerting.logs.levels }}
+          - alert: LogMessages{{ $level | title }}
+            expr: count_over_time(({namespace="{{ $.Release.Namespace }}", container="{{ include "generic-service.fullname" $ }}", {{ $.Values.alerting.logs.levelLabel }}="{{ $level }}"})[{{ $.Values.alerting.logs.countInterval }}]) > {{ $options.maxCount | default 0 }}
+            labels: {{- include "generic-service.alert-labels" $ | nindent 14 }} {{ $options.severity | default "warning" }}
+              topic: logs
+            annotations: {{- include "generic-service.alert-annotations" $ | nindent 14 }} {{ $level }} logs
+              description: '{{ $.Release.Name }} reported {{"{{ $value }}"}} {{ $level }} log messages in the last {{ $.Values.alerting.logs.countInterval }}.'
+          {{- end }}
+{{- end }}
+{{- if not .Values.alerting.logs.lokiSideCar }}
```

Comment:
Might be good to have a way to enable both the `ConfigMap` and the `LokiRule` at the same time for a transition period.

Reply:
I have introduced a new variable, `alerting.logs.createLokiRule` , with which we can allow the creation of both ConfigMap and LokiRule.